Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -725,3 +839,10 @@ extension ChannelPipeline.SynchronousOperations {
         }
     }
 }
+
+/// `NegotiationResult` is a generic negotiation result holder for HTTP/1.1 and HTTP/2
+@_spi(AsyncChannel)
+public enum NegotiationResult<HTTP1Output, HTTP2Output> {
+    case http1_1(HTTP1Output)
+    case http2(HTTP2Output)
+}
```

Comment:
I think we need to be a bit more specific here. Especially if we want to have something that does H1, H2, H3 in the future. What about naming this `HTTP1And2NegotiationResult`? Also open for better names cc @Lukasa @glbrntt

Reply:
I've gone with `NIONegotiatedHTTPVersion`